rest_framework/serializers.py

```diff
@@ -589,6 +589,11 @@ def __init__(self, *args, **kwargs):
         self.min_length = kwargs.pop('min_length', None)
         assert self.child is not None, '`child` is a required argument.'
         assert not inspect.isclass(self.child), '`child` has not been instantiated.'
+
+        if kwargs.get('instance', []) and kwargs.get('data', []):
+            assert len(kwargs.get("data", [])) == len(
+                kwargs.get("instance", [])), 'Data and instance should have same length'
```

Comment:
To avoid 2 times getting the dict value with default fallback, maybe use variables?

In python 3.8+ we are using `walrus :=` operator like this:
```suggestion
        if (instance := kwargs.get('instance', [])) and (data := kwargs.get('data', [])):
            assert len(data) == len(instance), 'Data and instance should have same length'
```

But to support previous versions of python, maybe we should do it as a separate variables:
```suggestion
        # instance, data = kwargs.get('instance', []), kwargs.get('data', [])  # if you prefer one line assignment support
        instance = kwargs.get('instance', [])
        data = kwargs.get('data', [])
        if instance and data:
            assert len(data) == len(instance), 'Data and instance should have same length'
```

Reply:
we have to support 3.6 now